http_types/utils.py

```diff
@@ -183,13 +183,15 @@ def from_dict(obj: Dict) -> Request:
             path = obj_copy["path"]
             obj_copy["pathname"] = parse_pathname(path)
 
+        bodyAsJson = obj_copy.get("bodyAsJson", None)
+
         if obj_copy.get("body", None) is None:
-            obj_copy["body"] = ""
+            obj_copy["body"] = json.dumps(bodyAsJson) if bodyAsJson else ""
 
         if obj_copy.get("headers", None) is None:
             obj_copy["headers"] = {}
 
-        if obj_copy.get("bodyAsJson", None) is None:
+        if bodyAsJson is None:
             body_as_json = parse_body(obj_copy["body"])
```

Comment:
Could you fix this line as well? We have requests where a body isn't json. It's customly fixed in test_runner.